app/src/androidTest/java/org/kiwix/kiwixmobile/utils/KiwixIdlingResource.kt

```diff
@@ -57,10 +55,10 @@ class KiwixIdlingResource : IdlingResource, IdleListener {
       if (kiwixIdlingResource == null) {
         kiwixIdlingResource = KiwixIdlingResource()
       }
-      if (kiwixIdlingResource != null) {
-        kiwixIdlingResource!!.idle = true
+      kiwixIdlingResource?.let {
+        it.idle = true
+        TestingUtils.registerIdleCallback(it)
       }
-      TestingUtils.registerIdleCallback(kiwixIdlingResource)
```

Comment:
I'm not sure what the intention was behind `kiwixidlingResource` being nullable or not. Do you think it's a good idea to make it non nullable?

Reply:
hi @gouri-panda .
it should be non null here because we are using ```getInstance()``` method to get ```KiwixIdlingResource``` and it's never return null.